syft/frameworks/torch/he/fv/context.py

```diff
@@ -12,14 +20,68 @@ class Context:
         rns_tool: A RNSTool class instance.
     """
 
-    def __init__(self, encryption_param):
+    def __init__(self, params):
 
-        self.param = encryption_param
+        # Validation of params provided for the encryption scheme.
+        self.validate(params)
+
+        self.param = params
+
+        self.rns_tool = RNSTool(params)
+
+    def validate(self, params):
+
+        # The number of coeff moduli is restricted to 62
+        if (
+            len(params.coeff_modulus) > COEFF_MOD_COUNT_MAX
+            or len(params.coeff_modulus) < COEFF_MOD_COUNT_MIN
+        ):
+            raise RuntimeError("Invalid coefficient modulus counts")
+
+        # Check for the range of coefficient modulus primes.
+        for i in range(len(params.coeff_modulus)):
+            if params.coeff_modulus[i] >> COEFF_MOD_BIT_COUNT_MAX or not (
+                params.coeff_modulus[i] >> (COEFF_MOD_BIT_COUNT_MIN - 1)
+            ):
+                raise RuntimeError("Invalid coefficient modulus values")
+
+            # Check for the relative prime of coefficient modulus primes
+            for j in range(i):
+                if math.gcd(params.coeff_modulus[i], params.coeff_modulus[j]) > 1:
+                    raise RuntimeError("Coefficient modulus are not pairwise relatively prime")
+
+        # Compute the product of all coeff moduli
+        total_coeff_modulus = reduce(lambda x, y: x * y, params.coeff_modulus)
+        # total_coeff_modulus_bit_count = get_significant_count(total_coeff_modulus)
```

Comment:
Shouldn't we also verify that the total_coeff_modulus is good for our n-bit security level?

Reply:
I left a todo there to implement it and it was the reason for WIP tag...😁

Reply:
I have added the code required for the security level checking. but It is currently commented with TODO because I would need to work on how to ask for the security level and if passed then it can be checked otherwise scheme will work without standard security.